anitya/db/models.py

```diff
@@ -338,6 +248,7 @@ class Project(Base):
 
     latest_version = sa.Column(sa.String(50))
     logs = sa.Column(sa.Text)
+    update_status = sa.Column(sa.Boolean, default=None)
```

Comment:
I imagine this will be something that is queried on, so it should probably have an index. Also, a nit, but the name is a bit confusing as a boolean: "update_status = False" doesn't mean much to me. "update_successful" or similar would be clearer. Does it need to be nullable?

One thing you could do is make this column an integer and store the HTTP status code. If things like DNS or TLS validation fail you could store a constant you define. Then you don't need the logs column here either.

Reply:
My idea was to use the `update_status` in this way:
- if `None` than update wasn't done yet - this should only happen on new project
- if `True` update was successful - this should be set if project was updated successfully, even if there is no new version retrieved
- if `False` update failed - this should be set if project check for new version failed for any reason

`project.logs` should contain error encountered when retrieving the version.

I wanted to name it `update` instead of `update_status`, but there is already method named `update`. But this could be easily changed to `update_successful`.

I don't think the HTTP status is good indicator. Most of the checks are failing before or after HTTP request is made. Like `Project is not correctly set-up` or `Can't parse new version succesfully`. These are the most frequent errors that raise `AnityaException` without any HTTP status code. I can probably map them to some specific status code and create a mapping for status codes to status messages.

What do you think about this?

Reply:
Hmm. I guess there are more possible errors than would be terribly fun to enumerate, and ``logs`` isn't indexed there's not a huge return on investment.

The only thing I'd think about is whether or not you want this column and also a column to track the last time a project was checked (so when the cron job runs you sort by oldest) or if those could be combined in some way.

This is only slightly related, but it doesn't make sense to me for a project to be created and not checked (I always check the "check now" box). Maybe we should force a check before creating the project. This way we will always report to the user their settings don't work and we don't have to handle the "project is created, but not checked" case.

Oh, and whatever you decide on it needs a docblock entry.

Reply:
The column containing last time the project was checked is already implemented here https://github.com/release-monitoring/anitya/pull/633.

But good idea about the automatic check of project. In this case, I just set the `update_status`, maybe I rename it to `update_successful`, to `False` until the check is actually successful.

Still it will be better to actually have some button `test` on create project site, that will try to check for new version, before commiting project to database and even better will be to do this also automatically in backend before commiting project.

Oh, I see I forgot to update docstring of Project class.

Reply:
Main issue with automatic check right now is that you need the project to exist in the database before check. This is the reason, why you can't do test check when creating the project.